internal/provider/resource_cluster.go

```diff
@@ -186,195 +178,158 @@ func clusterSubresourceSchema() *schema.Resource {
-	atLeastOneTypeOfDatastoreConfigured := false
-	if vsanDatastoreRaw, ok := object["vsan_datastore"]; ok && !validation_utils.IsEmpty(vsanDatastoreRaw) {
-		if len(vsanDatastoreRaw.([]interface{})) > 1 {
-			return nil, fmt.Errorf("more than one vsan_datastore config for cluster %q", clusterName)
-		}
-		vsanDatastoreListEntry := vsanDatastoreRaw.([]interface{})[0]
-		vsanDatastoreSpec, err := datastores.TryConvertToVsanDatastoreSpec(vsanDatastoreListEntry.(map[string]interface{}))
-		if err != nil {
-			return nil, err
-		}
-		atLeastOneTypeOfDatastoreConfigured = true
-		result.VSANDatastoreSpec = vsanDatastoreSpec
+func resourceClusterDelete(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcfClient := meta.(*SddcManagerClient)
+	apiClient := vcfClient.ApiClient
+
+	clusterUpdateParams := clusters.NewUpdateClusterParamsWithContext(ctx).
+		WithTimeout(constants.DefaultVcfApiCallTimeout)
+	clusterUpdateParams.ID = data.Id()
+	clusterUpdateSpec, _ := cluster.CreateClusterUpdateSpec(data, true)
+	clusterUpdateParams.SetClusterUpdateSpec(clusterUpdateSpec)
+
+	acceptedUpdateTask, _, err := apiClient.Clusters.UpdateCluster(clusterUpdateParams)
+	if err != nil {
+		return diag.FromErr(err)
 	}
-	if vmfsDatastoreRaw, ok := object["vmfs_datastore"]; ok && !validation_utils.IsEmpty(vmfsDatastoreRaw) {
-		if len(vmfsDatastoreRaw.([]interface{})) > 1 {
-			return nil, fmt.Errorf("more than one vmfs_datastore config for cluster %q", clusterName)
-		}
-		vmfsDatastoreListEntry := vmfsDatastoreRaw.([]interface{})[0]
-		vmfsDatastoreSpec, err := datastores.TryConvertToVmfsDatastoreSpec(vmfsDatastoreListEntry.(map[string]interface{}))
-		if err != nil {
-			return nil, err
-		}
-		atLeastOneTypeOfDatastoreConfigured = true
-		result.VmfsDatastoreSpec = vmfsDatastoreSpec
+	taskId := acceptedUpdateTask.Payload.ID
+	err = vcfClient.WaitForTaskComplete(ctx, taskId, false)
+	if err != nil {
+		return diag.FromErr(err)
 	}
-	if vsanRemoteDatastoreClusterRaw, ok := object["vsan_remote_datastore_cluster"]; ok && !validation_utils.IsEmpty(vsanRemoteDatastoreClusterRaw) {
-		if len(vsanRemoteDatastoreClusterRaw.([]interface{})) > 1 {
-			return nil, fmt.Errorf("more than one vsan_remote_datastore_cluster config for cluster %q", clusterName)
-		}
-		vsanRemoteDatastoreClusterListEntry := vsanRemoteDatastoreClusterRaw.([]interface{})[0]
-		vsanRemoteDatastoreClusterSpec, err := datastores.TryConvertToVSANRemoteDatastoreClusterSpec(
-			vsanRemoteDatastoreClusterListEntry.(map[string]interface{}))
-		if err != nil {
-			return nil, err
-		}
-		atLeastOneTypeOfDatastoreConfigured = true
-		result.VSANRemoteDatastoreClusterSpec = vsanRemoteDatastoreClusterSpec
+
+	clusterDeleteParams := clusters.NewDeleteClusterParamsWithContext(ctx).
+		WithTimeout(constants.DefaultVcfApiCallTimeout)
+	clusterDeleteParams.ID = data.Id()
+
+	acceptedDeleteTask, acceptedDeleteTask2, err := apiClient.Clusters.DeleteCluster(clusterDeleteParams)
```

Comment:
I am curious what are the two task return values here?

Reply:
Structs of type DeleteClusterOK (200) and DeleteClusterAccepted (202), that both have a single attribute "Payload" of type *models.Task. I've had bad experience with the Domain destroy task were I've wasted a whole run + env, wasting several hours because I've discarded the wrong return param, leading to nil pointer panic.

Reply:
In the Cluster Delete case only acceptedDeleteTask2 aka DeleteClusterAccepted is returned, DeleteClusterOK is nil, removed the first parameter.

Reply:
Got it! Thanks for the clarification.